### DISCUSSION TITLE
How to export ports to customized layers

### DISCUSSION THREAD

**Discussion by LIU-Yinyi** [gdsfactory/gdsfactory#2504] [Q&A] [answered]
By default, the `gdsfactory.add_pins` function would record the port information on the layers of **PORT**, **PORTE** of *generic_pdk*. 
Since my design covers the DC ports and RF ports, I want them to be exported separately onto **PORT_DC** and **PORT_RF** layers of my customized PDK. Besides, I also want to split components within the original layer of the device recognition boundary box **DEVREC** into two new layers: digital devices **DEVREC_D** and analogue devices **DEVREC_A**.

How can I do that for the best practice? Thanks.

**Comment by joamatab** [comment, marked as answer]
You can define those layers in your layermap and then 

add_pins(layer=layer_rf) for the functions that add pins with rf

and so on...
